### DISCUSSION TITLE
Going from nvim_buf_set_option to nvim_set_option_value

### DISCUSSION THREAD

**Discussion by jrock2004** [neovim/neovim#29681] [Q&A] [answered]
So I noticed that `nvim_buf_set_option` has been marked as deprecated. So how I used this today is in a autocmd to set mdx files as jsx files. I am trying to figure how to fix this deprecation

```
autocmd({ 'BufNewFile', 'BufRead' }, {
  pattern = { '*.mdx' },
  callback = function()
    local buf = vim.api.nvim_get_current_buf()

    vim.opt_local.spell = true
    vim.opt_local.wrap = true
    vim.api.nvim_buf_set_option(buf, 'filetype', 'jsx')
  end,
  desc = 'Working with mdx files',
})
```

So I was thinking `vim.api.nvim_set_option_value('filetype', 'jsx', {})` would be the answer but I think I might be missing something.

**Comment by zeertzjq** [comment, marked as answer]
It should be `vim.api.nvim_set_option_value('filetype', 'jsx', { buf = buf })`

**Reply by jrock2004** [reply]
Thank you
